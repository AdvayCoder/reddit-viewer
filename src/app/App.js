import React, { useEffect } from 'react';
import './App.css';
import { BrowserRouter as Router, Routes, Route} from 'react-router-dom';
import NavBar from '../Components/NavBar/NavBar';
import HomePage from '../pages/HomePage/HomePage';
import InitFonts from './InitFonts/InitFonts';
import { useDispatch } from 'react-redux';
import {loadPosts} from '../features/posts/postsSlice';

function App() {

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadPosts('https://www.reddit.com/r/ProgrammerHumor.json'));
  }, [dispatch])

  return (
    <Router>
      <InitFonts />
      <NavBar />
      <Routes>
        <Route path="/" element={ <HomePage /> } />
      </Routes>
    </Router>
  );
}

export default App;
