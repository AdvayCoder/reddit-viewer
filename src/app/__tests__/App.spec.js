import React from 'react';
// import { render } from '@testing-library/react';
// import { Provider } from 'react-redux';
// import { store } from '../store';
import App from '../App';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

configure({ adapter: new Adapter() })

describe('App Component', () => {

  it('returns jsx', () => {
    const wrapper = shallow(
      <App />
    )

    expect(wrapper).toMatchSnapshot();
  })
  
});