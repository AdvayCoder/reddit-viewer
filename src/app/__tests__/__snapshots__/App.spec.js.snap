// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component returns jsx 1`] = `ShallowWrapper {}`;
